### DISCUSSION TITLE
If you want to filter a table based on a child table's values Typescript does not help and shows Error?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#5262] [Questions]
Here is an example from supabase doc --
```
const { data, error } = await supabase
  .from('messages')
  .select('*, users!inner(*)')
  .eq('users.username', 'Jane')
```
Using this with Typescript type something like 
```
interface Message{
...
users:Users {username:string}
}
```
and then
```
const { data, error } = await supabase
  .from<Message>('messages')
  .select('*, users!inner(*)')
  .eq('users.username', 'Jane')
```
does not work and shows type error. 
Can u please give me a working example along with the case when **"Query the same foreign table multiple times"** if possible.

**Comment by DanMossa** [comment]
I'm having the same issue. Did you figure out how to write your query?
